### PR TITLE
feat: pull appointment caregiver from general

### PR DESCRIPTION
**Pull request by fedasevich**
## Describe your changes

- Pulled fixes from general branch
- Created reusable caregiver drawer
- Added timezone detection

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/CC-79?atlOrigin=eyJpIjoiMDRlODJjMGFkMTE5NDg3YzkwYWRjYmU5N2VhMWQzNmIiLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by Lilly-nn — APPROVED** [review]

File: src/components/complete-profile-second/work-form/validation.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { ObjectSchema, boolean, date, object, string } from 'yup';
-import { useLocales } from 'src/locales';
-import { MAX_CHARACTERS_LENGTH } from 'src/constants';
 import { CompleteProfileSecondValues } from 'src/components/complete-profile-second/types';
+import { MAX_CHARACTERS_LENGTH } from 'src/constants';
+import { useLocales } from 'src/locales';
+import { ObjectSchema, boolean, date, object, string } from 'yup';
```
Comment (ProggerOleg) [inline src/components/complete-profile-second/work-form/validation.ts:4]: 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path

File: src/components/create-appointment-fourth/places-autocomplete/helpers.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { PLACES_DETAILS_FIELDS } from 'src/components/create-appointment-fourth/places-autocomplete/constants';
 import { AddressComponent } from 'src/components/create-appointment-fourth/places-autocomplete/enums';
 import { AutocompletedLocation } from 'src/components/create-appointment-fourth/types';
-import { getDetails } from 'use-places-autocomplete';
+import { GeocodeResult, getDetails, getLatLng } from 'use-places-autocomplete';
```
Comment (ProggerOleg) [inline src/components/create-appointment-fourth/places-autocomplete/helpers.ts:4]: same

File: src/components/reusable/drawer/caregiver-drawer/CaregiverDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+          {selectedCaregiver.firstName} {selectedCaregiver.lastName}
+        </DrawerTitle>
+      </DrawerHeader>
+      <DrawerStats flexDirection="row">
+        <DrawerAvatar
+          src={getMockCaregiverAvatar(BIG_CAREGIVER_AVATAR_SIZE)}
+          alt={`${selectedCaregiver.firstName} ${selectedCaregiver.lastName}`}
+        />
+        <Stack alignItems="center" flexDirection="row" justifyContent="space-evenly" width="100%">
+          <Stack flexDirection="column" alignItems="center">
```
Comment (ProggerOleg) [inline src/components/reusable/drawer/caregiver-drawer/CaregiverDrawer.tsx:80]: It's not really inline styles, maybe ok if 1-2, but here just to many. Would be a better practise to create styled component

File: src/redux/api/timezoneApi.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+export const timezoneApi = createApi({
+  reducerPath: 'timezoneApi',
+  baseQuery: fetchBaseQuery({
+    baseUrl: GOOGLE_MAP_API,
+  }),
+  endpoints: (builder) => ({
+    getTimezone: builder.query<TimeZoneInfo, TimezonePayload>({
+      query: (params) => ({ url: `/timezone/json`, method: 'GET', params }),
```
Comment (ProggerOleg) [inline src/redux/api/timezoneApi.ts:26]: please move url to routes file

File: src/components/complete-profile-second/work-form/validation.ts (hunk 1 above)
Comment (fedasevich) [reply src/components/complete-profile-second/work-form/validation.ts]: fixed

File: src/components/reusable/drawer/caregiver-drawer/CaregiverDrawer.tsx (hunk 3 above)
Comment (fedasevich) [reply src/components/reusable/drawer/caregiver-drawer/CaregiverDrawer.tsx:81]: fixed

File: src/redux/api/timezoneApi.ts (hunk 4 above)
Comment (fedasevich) [reply src/redux/api/timezoneApi.ts]: fixed

File: src/redux/api/appointmentApi.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  endpoints: (builder) => ({
+    createAppointment: builder.mutation<void, AppointmentPayload>({
+      query: (body) => ({
+        url: route.appointment,
+        method: 'POST',
+        body,
+      }),
+    }),
+    getFilteredCaregivers: builder.query<PreviewCaregiver[], URLSearchParams>({
+      query: (params) => ({ url: `${route.caregivers}/filter`, method: 'GET', params }),
```
Comment (nadiia-tsytsylina) [inline src/redux/api/appointmentApi.ts:55]: here better to add 'filter' to routes

**Review by nadiia-tsytsylina — APPROVED** [review]

**Review by ProggerOleg — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**